sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java

```diff
@@ -660,13 +1113,42 @@ public CarbonWriter build() throws IOException, InvalidLoadOptionException {
       // removed from the load. LoadWithoutConverter flag is going to point to the Loader Builder
       // which will skip Conversion Step.
       loadModel.setLoadWithoutConverterStep(true);
-      return new AvroCarbonWriter(loadModel, hadoopConf, this.avroSchema);
+      AvroCarbonWriter avroCarbonWriter = new AvroCarbonWriter(loadModel,
+          hadoopConf, this.avroSchema);
+      if (!StringUtils.isEmpty(filePath)) {
+        avroCarbonWriter.setDataFiles(this.dataFiles);
+      }
+      return avroCarbonWriter;
     } else if (this.writerType == WRITER_TYPE.JSON) {
       loadModel.setJsonFileLoad(true);
-      return new JsonCarbonWriter(loadModel, hadoopConf);
+      JsonCarbonWriter jsonCarbonWriter = new JsonCarbonWriter(loadModel, hadoopConf);
+      if (!StringUtils.isEmpty(filePath)) {
+        jsonCarbonWriter.setDataFiles(this.dataFiles);
+      }
+      return jsonCarbonWriter;
+    } else if (this.writerType == WRITER_TYPE.PARQUET) {
+      loadModel.setLoadWithoutConverterStep(true);
+      AvroCarbonWriter avroCarbonWriter = new AvroCarbonWriter(loadModel,
+          hadoopConf, this.avroSchema);
+      ParquetCarbonWriter parquetCarbonWriter = new
+          ParquetCarbonWriter(avroCarbonWriter, hadoopConf);
```

Comment:
Instead of creating instance of `AvroCarbonWriter` and passing it to `ParquetCarbonWriter`, create `AvroCarbonWriter` instance internally within the constructor of `ParquetCarbonWriter` Suggest same for `ORCCarbonWriter` below

Reply:
done